<?xml version="1.0"?>
<!--XML-->
<launch>
 
    <arg name="gui" default="False" />
    <arg name="use_tf_static" default="false"/> 
    <param name="use_gui" value="$(arg gui)"/>
    <arg name="marker_size" default="9" />
    <arg name="max_new_marker_error" default="0.08" />
    <arg name="max_track_error" default="0.0" />
    <arg name="cam_image_topic" default="/stereo/left/image_rect" />
    <arg name="cam_info_topic" default="/stereo/left/camera_info" />
    <arg name="depth" value="false"/>
    <arg name="bundle" value="true"/>
    <arg name="output_frame" default="/base_link" />
    <arg name="med_filt_size" default="10" />
    <arg name="bundle_files" default="$(find sr_description_common)/ar_markers/ar_marker_0_1_2_3_a4.xml" />

    <arg name="marker_to_root_cartesian" default="[-0.12, 0.67, 0.0, 0.0, 0.0, -1.5707]"/>
    <arg name="marker_frame_name" default="ar_marker_0"/>
    <arg name="root_frame_name" default="ra_base_link"/>
    <arg name="camera_frame_name" default="base_link"/>
    <arg name="window_width" default="2000"/>
    <arg name="filtering" default="true"/>
    <arg name="continuous" default="true"/>
    <param name="robot_description" command="$(find xacro)/xacro '$(find deimos)/urdf/standalone.urdf.xacro'" />

    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" ></node>
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher">
	    <param name="use_tf_static" value="$(arg use_tf_static)"/>
    </node>
     
    <group ns="stereo">
        <node pkg="deimos" type="tara_node" name="deimos" output="screen" required="true">
            <param name="width" type="int" value="752" />
            <param name="height" type="int" value="480" />
            <param name="fps" type="int" value="60" />
            <param name="device" type="string" value="/dev/video0" />
            <param name="exposureValue" type="int" value="1" />
            <param name="cameraLeft_info_url" type="string" value="file://${ROS_HOME}/camera_info/left.yaml" />
            <param name="cameraRight_info_url" type="string" value="file://${ROS_HOME}/camera_info/right.yaml" />
        </node>

        <node pkg="stereo_image_proc" type="stereo_image_proc" name="stereo_image_proc">
	        <!-- BM, for some reason SGBM fails -->
	        <param name="stereo_algorithm" value="0" />
	        <param name="min_disparity" value="4" />
	        <param name="disparity_range" value="128" />
            <param name="correlation_window_size" value="35" />
	    </node>
        <include file="$(find sr_camera_pose_estimation)/launch/mock_world_transforms.launch" />

	    <node pkg="tf2_ros" type="static_transform_publisher" name="left_image_broadcaster" args="0 0 0 0 0 0 1 cameraLeft_optical left/image_raw" />
    </group>

    <node name="ar_track_alvar" pkg="ar_track_alvar" type="findMarkerBundlesNoKinect" respawn="false" output="screen" args="$(arg marker_size) $(arg max_new_marker_error) $(arg max_track_error) $(arg cam_image_topic) $(arg cam_info_topic) $(arg output_frame) $(arg bundle_files)" />    

    <node pkg="sr_camera_pose_estimation" type="camera_transform_publisher.py" name="camera_transform_publisher" output="screen">
        <rosparam param="marker_to_root_cartesian" subst_value="True">$(arg marker_to_root_cartesian)</rosparam>
        <param name="marker_frame_name" value="$(arg marker_frame_name)" />
        <param name="root_frame_name" value="$(arg root_frame_name)" />
        <param name="camera_frame_name" value="$(arg camera_frame_name)" />
        <param name="window_width" value="$(arg window_width)" />
        <param name="filtering" value="$(arg filtering)" />
        <param name="continuous" value="$(arg continuous)" />
    </node>

</launch>